pkgs/tools/security/gnupg/21.nix

```diff
@@ -23,9 +23,14 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [
-    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo
-    readline libusb gnutls adns openldap zlib bzip2
-  ];
+    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo ]
+      ++ optional (adns     != null) adns
```

Comment:
Checking whether a  variable is `null` or not before adding it to the `buildInputs` is nice, no doubt, but it has no effect on the actual build. You can add `null` to `buildInputs` and it doesn't do any harm. Which makes me wonder whether it's worthwhile to add those checks?